packages/go_router/example/lib/shell_route.dart

```diff
@@ -0,0 +1,222 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+final GlobalKey<NavigatorState> _rootNavigatorKey = GlobalKey<NavigatorState>();
+
+void main() {
+  runApp(ShellRouteExampleApp());
+}
+
+/// An example demonstrating how to use [ShellRoute]
```

Comment:
I'm worried that folks will see this example and think you need to define a root navigator key to use `ShellRoute`. I wonder if we should have two separate shell route examples:

1. A basic shell route example
2. A shell route example that shows you how to "break out" of the shell route and display on top of the root navigator.

Reply:
SGTM - I'll break it out into two samples